modules/ting_reference/ting_reference.module

```diff
@@ -558,28 +560,23 @@ function _ting_reference_autocomplete($string) {
 
   // Only use auto-complete if more that 3 chars have been written.
   if ($string && (strlen($string) > 3)) {
-    // Try to build the best search query based on the inputted string.
-    if (preg_match('/(^\d+$)|(^\d+:\d+$)|(=)/', $string)) {
-      // The query is either ISBN, tid or CQL search expression.
-      $query = $string;
-    }
-    else {
-      // Add wildcard to the search string.
-      $query  = _ting_search_quote($string) . '*';
-    }
+    // Build a search-query and execute it.
+    $search_query = ting_start_query();
+    $search_query->setSimpleQuery($string);
+    $search_query->setCount(10);
+    $search_query->setFacets([]);
```

Comment:
If we have a fluid interface for the search query perhaps this can be made more compact.